### PR TITLE
[SDP-1096] dev: create the topics if not exists on kafka startup

### PR DESCRIPTION
**Pull request by CaioTeixeira95**
### What

Create Kafka's topic on the container startup.

### Why

Avoid concurrent problems when reading from the topics.

### Known limitations

N/A

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [ ] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: dev/docker-compose-sdp-anchor.yml [hunk 1]
```diff
@@ -194,6 +194,25 @@ services:
       - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://kafka:9092,EXTERNAL://localhost:9094
       - KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=CONTROLLER:PLAINTEXT,EXTERNAL:PLAINTEXT,PLAINTEXT:PLAINTEXT
       - KAFKA_CFG_CONTROLLER_LISTENER_NAMES=CONTROLLER
+    healthcheck:
+      test: /opt/bitnami/kafka/bin/kafka-topics.sh --bootstrap-server kafka:9092 --list || exit -1
```
Comment (marwen-abid) [inline dev/docker-compose-sdp-anchor.yml:198]: nit: I believe you can just use `kafka-topics.sh --bootstrap-server kafka:9092 --list || exit -1`

File: dev/docker-compose-sdp-anchor.yml [hunk 2]
```diff
@@ -194,6 +194,25 @@ services:
...
+
+  kafka-init:
+    image: docker.io/bitnami/kafka:3.6
+    entrypoint: [ "/bin/bash", "-c" ]
+    depends_on:
+      kafka:
+        condition: service_healthy
+    command: |
+      "
+        /opt/bitnami/kafka/bin/kafka-topics.sh --create --if-not-exists --topic events.receiver-wallets.new_invitation --bootstrap-server kafka:9092
```
Comment (marwen-abid) [inline dev/docker-compose-sdp-anchor.yml:212]: nit: Similar here, I believe we can just use `kafka-topics.sh --create --if-not-exists --topic events.receiver-wallets.new_invitation --bootstrap-server kafka:9092`

**Review by marwen-abid — APPROVED** [review]
LGTM! This is amazing

File: dev/docker-compose-sdp-anchor.yml [hunk 3]
```diff
@@ -194,6 +194,25 @@ services:
...
+        /opt/bitnami/kafka/bin/kafka-topics.sh --create --if-not-exists --topic events.payment.payment_completed --bootstrap-server kafka:9092
+        /opt/bitnami/kafka/bin/kafka-topics.sh --create --if-not-exists --topic events.payment.ready_to_pay --bootstrap-server kafka:9092
+      "
```
Comment (marcelosalloum) [inline dev/docker-compose-sdp-anchor.yml:215]: Hey @CaioTeixeira95, could you plz add comments here and in [internal/events/handler.go](https://github.com/stellar/stellar-disbursement-platform-backend/blob/f8e0abd694b5b9b433d78809d4d15a1a5520ba84/internal/events/handler.go#L7-L12) reminding devs to keep these topics synchronized between the Go code and the docker-compose file?

**Review by marcelosalloum — APPROVED** [review]
Awesome, thanks so much for doing that!

Plz, feel free to merge it after addressing the comments below.

File: dev/docker-compose-sdp-anchor.yml (hunk 3 above)
Comment (CaioTeixeira95) [reply dev/docker-compose-sdp-anchor.yml]: Could you please take a look if the comment is ok?

PTAL - [15150a9](https://github.com/stellar/stellar-disbursement-platform-backend/pull/200/commits/15150a9070ba7a29be1cb7050c0b0d568b7f341f)

**Review by marcelosalloum — APPROVED** [review]
LGTM! Let's Go 🚀

**Pull request merged**